runtime/plugins/comment/comment.lua

```diff
@@ -150,22 +150,36 @@ function comment(bp, args)
     local lines = {}
     local curData = {}
     -- gather cursor data and lines to (un)comment
-    for i = 0,#bp.Buf:getCursors()-1 do
-        local cursor = bp.Buf:getCursor(i)        
+    for i = 1,#bp.Buf:getCursors() do
+        local cursor = bp.Buf:getCursor(i-1)
         local hasSelection = cursor:HasSelection()
-        table.insert(curData, {
-            sel = -cursor.CurSelection,
-            curpos = -cursor.Loc,
-            cursor = cursor,
-            hasSelection = hasSelection
-        })
+        local staticEnd = false
+        local startSel = 1
+        local endSel = 2
         if hasSelection then
-            for lineN = cursor.CurSelection[1].Y, cursor.CurSelection[2].Y do
+            if cursor.CurSelection[1]:GreaterThan(-cursor.CurSelection[2]) then
+                startSel = 2
+                endSel = 1
+            end
+            for lineN = cursor.CurSelection[startSel].Y, cursor.CurSelection[endSel].Y do
                 lines[lineN] = true
             end
+            if cursor.CurSelection[endSel].X == 0 then
+                lines[cursor.CurSelection[endSel].Y] = nil
+                staticEnd = true
+            end
```

Comment:
This can end up overwriting `lines` that were already set to true by another cursor.

Demo: https://asciinema.org/a/VB99WIg0Oo6HJAkqc4KVeHEPP

Reply:
Well caught! I have merged your suggestions. It solves the bug, and didn't seem to change any other behavior.